pkg/kube/client.go

```diff
@@ -484,13 +484,30 @@ type UpdateOptions struct {
 func (c *Client) UpdateWithOptions(namespace string, originalReader, targetReader io.Reader, opts UpdateOptions) error {
 	original, err := c.BuildUnstructured(namespace, originalReader)
 	if err != nil {
-		return fmt.Errorf("failed decoding reader into objects: %s", err)
+		// Checking for removed Kubernetes API error so can provide a more informative error message to the user
+		// Ref: https://github.com/helm/helm/issues/7219
+		if strings.Contains(err.Error(), "unable to recognize \"\": no matches for kind") {
```

Comment:
Can we make `"unable to recognize \"\": no matches for kind"` a constant since used twice?

Reply:
Done